index.js

```diff
@@ -268,16 +268,34 @@ function processFormats(formats) {
   //  - play/pausing and seeking in play content still works
 
   const tracks = []
-  for (const quality of [2160, 1080, 720]) {
-    tracks.push(filteredFormats.find((format) => (format.height === quality && format.acodec === 'none')));
-    tracks.push(filteredFormats.find((format) => (format.height === quality && format.acodec !== 'none')));
-  }
 
-  // (temporary)
-  // Vivi Anywhere cannot (yet) play m3u8 or split tracks.
-  if (!tracks.find((format) => (format && (format.acodec !== 'none' && format.protocol === 'https')))) {
-    // None of the previously selected tracks are combined non-m3u8 tracks.... so find the best combined non-m3u8 track now
-    tracks.push(filteredFormats.find((format) => (format.acodec !== 'none' && format.protocol === 'https')));
+  if (isStream) {
+    // Livestreams will always have a combined m3u8 track, return this.
+    // (For a livestream, ALL its tracks are m3u8. This means if we decide to return split tracks for
+    // a livestream, it will be a m3u8 audio track and a m3u8 video track.)
+    for (const quality of [2160, 1080, 720]) {
+      tracks.push(filteredFormats.find((format) => (format.height === quality && format.acodec !== 'none')));
+    }
+  } else {
+    // Non-livestreams
+        
+    // Find the best split track for each quality level
+    for (const quality of [2160, 1080, 720]) {
+      tracks.push(filteredFormats.find((format) => (format.height === quality && format.acodec === 'none')));
+    }
+
+    // VA can only play non-m3u8 combined tracks. Find the best track that VA can play
+    const vaTrack = filteredFormats.find((format) => (format.height <= 1080 && format.acodec !== 'none' && format.protocol === 'https'));
```

Comment:
After the new split track changes, VA will be able to play split tracks with m3u8 video tracks just not audio tracks. But this if fine for now.

Reply:
yeah, we are going to have to talk about this sorry. But that's a separate problem. I'll post an update into ytdl-yips soon once I get to the bottom of this triage ticket that just came in